lib/azure/resource_management/ARM_interface.rb

```diff
@@ -716,6 +725,13 @@ def create_vm_extension(params)
         vm_extension
       end
 
+      def extension_already_installed?(server)
+        server.resources.each do |extension|
+          return true if extension.properties.publisher == "Chef.Bootstrap.WindowsAzure"

```

Comment:
@NimishaS In future it may happen that same publisher publishes some another tool for azure extension. So can we use chef_extension_name value like `LinuxChefClient` for `Linux` VMs and `ChefClient` for `Windows` VMs to check whether Chef Extension is already installed or not on the target VM.